internal/validation/validation.go

```diff
@@ -701,22 +702,50 @@ func validateLiteral(c *opContext, l common.Literal) {
 	}
 }
 
+func stringLiteral(value string) common.Literal {
+	return &common.BasicLit{Type: scanner.String, Text: value}
+}
+
 func parseLiteral(value string) common.Literal {
+	if value == "" {
+		return stringLiteral("")
+	}
 	lexer := common.NewLexer(value, false)
-	lexer.ConsumeWhitespace()
-	return common.ParseLiteral(lexer, false)
+	if unicode.IsSpace(lexer.Advance()) {
+		// assuming values of other scalar types don't start with spaces
+		return stringLiteral(value)
+	}
+	rv := common.ParseLiteral(lexer, true)
+	if lexer.Peek() != scanner.EOF {
+		// if it can be parsed as several simple values it is assumed to be a string
+		return stringLiteral(value)
+	}
+	return rv
 }
 
-func varBinding(c *opContext, name string) (common.Literal, bool) {
+func varBinding(c *opContext, name string) (rv common.Literal, success bool) {
 	if binding, ok := c.variables[name]; ok {
 		if literal, ok := binding.(common.Literal); ok {
 			return literal, true
 		}
+
+		defer func() {
```

Comment:
Let's drop it as it was before, panic means internal problem in the library to fix, to be handled by the application (which is already being done). Whether it panics or just fails to parse a valid request does not matter from the app perspective - it still fails.